### PR TITLE
MM1 style transmission.

### PR DESCRIPTION
**Pull request by Wanted9811**

### CODE REVIEW

File: src/modules/vehicle/drivetrain.h [hunk 1]
```diff
@@ -4,13 +4,40 @@
 namespace MM2
 {
     // Forward declarations
-
+    class vehDriveTrain;
```
Comment (Dummiesman) [inline src/modules/vehicle/drivetrain.h]: The class name should be `vehDrivetrain` (no capital T on train)

File: src/modules/vehicle/drivetrain.h [hunk 2]
```diff
@@ -4,13 +4,40 @@
...
+
+        static void BindLua(LuaState L) {
+            LuaBinding(L).beginExtendClass<vehDriveTrain, asNode>("vehDriveTrain")
+                .addFunction("Attach", &Attach)
+                .addFunction("Detach", &Detach)
+            .endClass();
+        }
+    };
+
+    ASSERT_SIZEOF(vehDriveTrain, 0x20);
```
Comment (Dummiesman) [inline src/modules/vehicle/drivetrain.h]: sizeof(vehDrivetrain) is 4C. Add in the rest of the members or fill the remaining space with a buffer like seen in other classes.

File: src/modules/vehicle/engine.h [hunk 3]
```diff
@@ -40,6 +41,7 @@ namespace MM2
         Matrix34 *EngineVisualMatrixPtr;
         vehCarSim *VehCarSimPtr;
         vehTransmission *VehTransmissionPtr;
+        vehDriveTrain *VehDriveTrainPtr;
```
Comment (Dummiesman) [inline src/modules/vehicle/engine.h]: Where does this come from? vehEngine appears to have a size of 0x84 (you can see by how it's initialized)

**Review by Dummiesman — CHANGES_REQUESTED** [review]

File: src/modules/vehicle/drivetrain.h (hunk 1 above)
Comment (Wanted9811) [reply src/modules/vehicle/drivetrain.h]: I will change that, thanks for notifying me!

**Pull request merged**